nx.json

```diff
@@ -3,9 +3,46 @@
     "default": {
       "runner": "nx-cloud",
       "options": {
-        "cacheableOperations": [],
+        "cacheableOperations": [
+          "lint"
+        ],
         "accessToken": "ZmNlNjA0YzAtNTM1NS00MDIwLWFlMWItNWYxYzNiMjQ4N2VkfHJlYWQtb25seQ=="
       }
     }
+  },
+  "targetDefaults": {
+    "lint": {
+      "inputs": [
+        "default",
+        "{workspaceRoot}/.eslintrc.js"
+      ]
+    },
+    "build": {
+      "dependsOn": [
+        "^build"
+      ]
+    },
+    "build-prod": {
+      "dependsOn": [
+        "^build-prod"
+      ]
+    }
+  },
+  "namedInputs": {
+    "sharedGlobals": [],
+    "default": [
+      "{projectRoot}/**/*",
+      "sharedGlobals"
+    ],
+    "production": [
+      "default",
+      "!{projectRoot}/**/*.spec.ts",
+      "!{projectRoot}/**/*.md",
+      "!{projectRoot}/tsconfig.spec.json",
+      "!{projectRoot}/.eslintrc.json",
+      "!{projectRoot}/.mocharc.{js,json}",
+      "!{projectRoot/cypress.config.{ts,js}",
+      "!{projectRoot/**/*.cy.ts"
+    ]
```

Comment:
@jordanpowell88 I am not sure how NX works exactly, so I am not sure how to validate the changes here in `nx.json`.  Can you describe what each part is doing and point to NX documentation that describes how these should be configured.  I am guessing it is describing what should or should not be cache for the task, but not sure.

Reply:
Thanks for the explanation.  Final question in this section... Is the `production` option under `namedInputs` being used by anything?

Reply:
Not currently. But I went ahead and added what I would expect to be starting values we would use. We can certainly remove this if we want for the time being though